miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditWorksetController.java

```diff
@@ -0,0 +1,68 @@
+package uk.ac.bbsrc.tgac.miso.webapp.controller;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.acls.model.NotFoundException;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.ModelAndView;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Workset;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.service.WorksetService;
+
+@Controller
+@RequestMapping("/workset")
+public class EditWorksetController {
+
+  private static final String JSP = "/pages/editWorkset.jsp";
+
+  private static final String MODEL_ATTR_TITLE = "title";
+  private static final String MODEL_ATTR_PAGEMODE = "pageMode";
+  private static final String MODEL_ATTR_ID = "worksetId";
+  private static final String MODEL_ATTR_JSON = "worksetJson";
+  private static final String MODEL_ATTR_SAMPLES = "samples";
+  private static final String MODEL_ATTR_LIBRARIES = "libraries";
+  private static final String MODEL_ATTR_DILUTIONS = "dilutions";
+
+  @Autowired
+  private WorksetService worksetService;
+
+  @ModelAttribute("maxLengths")
+  public Map<String, Integer> getColumnSizes() throws IOException {
+    return worksetService.getColumnSizes();
+  }
+  
+  @RequestMapping("/new")
+  public ModelAndView newArray(ModelMap model) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)